cypress/e2e/registration.cy.js

```diff
@@ -1,11 +1,84 @@
 /// <reference types='cypress' />
 
 describe('Student Registration page', () => {
-  before(() => {
+  let user;
 
+  before(() => {
+    cy.visit('/');
+    cy.task('generateUser').then((generateUser) => {
+      user = generateUser;
+    });
   });
 
-  it('', () => {
+  it('should register a new student', () => {
+    cy.findByPlaceholder('First Name')
+      .type(user.firstName);
+
+    cy.findByPlaceholder('Last Name')
+      .type(user.lastName);
+
+    cy.get('#userEmail').type(user.email);
+
+    cy.contains('.custom-control-label', user.gender)
+      .click();
+
+    cy.findByPlaceholder('Mobile Number')
+      .type(user.mobileNumber);
+
+    cy.get('#dateOfBirthInput')
+      .type('{selectAll}15 Sep 2005{enter}');
+
+    cy.get('.subjects-auto-complete__value-container')
+      .type('m{enter}' + 'e{enter}' + 'p{enter}');
+
+    cy.contains('.custom-control-label', user.hobby)
+      .click();
+
+    cy.findByPlaceholder('Current Address')
+      .type(user.address);
+
+    cy.get('#state')
+      .type('{downArrow}{enter}');
+
+    cy.get('#city')
+      .type('{downArrow}{enter}');
```

Comment:
The state and city are being selected by using keyboard input instead of explicitly selecting a value. This could cause the test to fail if the order of options in these dropdowns changes.